agent/php_internal_instrument.c

```diff
@@ -1527,10 +1560,63 @@ NR_INNER_WRAPPER(mysqli_stmt_prepare) {
  */
 NR_INNER_WRAPPER(memcache_function) {
   nr_php_instrument_datastore_operation_call(nr_wrapper, NR_DATASTORE_MEMCACHE,
-                                             nr_wrapper->extra, NULL,
+                                             nr_wrapper->extra, NULL, false,
                                              INTERNAL_FUNCTION_PARAM_PASSTHRU);
 }
 
+NR_INNER_WRAPPER(memcached_connect_function) {
+  char* host = NULL;
+  nr_string_len_t host_len = 0;
+  zend_long port = 0;
+  zend_long weight = 0;
+  nr_datastore_instance_t* instance = NULL;
+
+  if (SUCCESS
+      == zend_parse_parameters_ex(
+          ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS() TSRMLS_CC, "s|ll", &host,
+          &host_len, &port, &weight)) {
+    if (NULL != host) {
+      instance = nr_php_memcached_create_datastore_instance(host, port);
+    }
+  }
+  nr_php_instrument_datastore_operation_call(nr_wrapper, NR_DATASTORE_MEMCACHE,
+                                             NULL, instance, true,
+                                             INTERNAL_FUNCTION_PARAM_PASSTHRU);
+}
+
+NR_INNER_WRAPPER(memcached_multi_connect_function) {
+  zval* servers = NULL;
+  zval* server = NULL;
+  zend_ulong num_key = 0;
+  nr_php_string_hash_key_t* string_key = NULL;
+  nr_datastore_instance_t* instance = NULL;
+  int zcaught = 0;
+
+  if (SUCCESS
+      == zend_parse_parameters_ex(
+          ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS() TSRMLS_CC, "a", &servers)) {
+    ZEND_HASH_FOREACH_KEY_VAL(Z_ARRVAL_P(servers), num_key, string_key, server) {
+      (void)num_key;
+      (void)string_key;
```

Comment:
If the `num_key` and `string_key` are not needed/used, then use `ZEND_HASH_FOREACH_VAL`:
```suggestion
    ZEND_HASH_FOREACH_VAL(Z_ARRVAL_P(servers), server) {
```